src/main/scala/supporters/ExpressionTranslator.scala

```diff
@@ -98,9 +98,25 @@ trait ExpressionTranslator {
           }
         )))
 
-        val weight = sourceQuant.info match {
-          case w: WeightedQuantifier => Some(w.weight)
-          case _ => None
+        val weight = sourceQuant.info.getUniqueInfo[WeightedQuantifier] match {
+          case Some(w) => Some(w.weight)
+          case None => sourceQuant.info.getUniqueInfo[AnnotationInfo] match {
+            case Some(ai) if ai.values.contains("weight") =>
+              ai.values("weight") match {
+                case Seq(w) =>
+                  try {
+                    Some(w.toInt)
```

Comment:
Zero seems to be fine, yes. With negative values you get a Z3 error apparently. 
I didn't handle that case because the existing code for ``WeightedQuantifier`` doesn't handle it either, and any problems happen on the SMT level. E.g., if CVC5 somehow has a meaning for negative weight, Silicon would correctly give it the specified weight and it would interpret it however it wants. In Z3's case it turns out it crashes.

Reply:
But I don't have any strong opinions really. If you think we should reject negative weights I'll add a check (also for the existing ``WeightedQuantifier``).

Reply:
Maybe it's a Silicon/Carbon thing, but I think it'd be a good principle that all Viper programs that pass type-checking and consistency checks do not crash, but report verification errors in the worst case.

(I'm not sure how the Z3 crash is reported in this PR.)

Reply:
It's not reported nicely, you get an exception.

Yeah I agree that'd be nice. Then IMO we should reject ``WeightedQuantifier`` annotations with negative values, too,

Reply:
Done, negative values are rejected for both kinds of Info objects.